x-pack/plugins/enterprise_search/public/applications/app_search/components/data_panel/data_panel.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+
+interface Props {
+  title: string;
+  subtitle?: string;
+  iconType?: string;
+  action?: React.ReactNode;
+  filled?: boolean;
+}
+
+export const DataPanel: React.FC<Props> = ({
+  title,
+  subtitle,
+  iconType,
+  action,
+  filled,
+  children,
+}) => (
+  <EuiPanel color={filled ? 'subdued' : 'plain'} hasShadow={false}>
+    <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+      <EuiFlexItem>
+        <EuiFlexGroup>
+          {iconType ? (
+            <EuiFlexItem>
+              <EuiIcon data-test-subj="dataPanelIcon" type={iconType} />
+            </EuiFlexItem>
+          ) : null}
+          <EuiFlexItem>
+            <EuiTitle data-test-subj="dataPanelTitle" size="xs">
+              <h4>{title}</h4>
```

Comment:
I'm very concerned about the blanket setting of a `h4` tag here. Semantically and accessibly, that's not how heading hierarchies should go - there should be a clear and incremental increase of h1 -> h2 -> h3 -> h4 titles/sections, we can't just skip from h1 -> h4 - that will usually trigger an axe warning.

I think we should change `title` to a ReactNode instead of a string and have components pass in their own `<h2>title</h2>` or `<h4>title</h4>` so pages can determine their own heading levels/hierarchies.

Reply:
Also I'm super available tomorrow if you'd prefer to pair on change requests together!

Reply:
I hear ya. I think I'll go with an `EuiTitle` component instead. I want the size to be consistent, but agree that this isn't the most semantic way of handling this.

Reply:
Updated: https://github.com/elastic/kibana/pull/92939/commits/a98160c802d311dfe56a2c06243479e9ff54bfbf#diff-8f297480eb034e309622e67ac786f4ce2b745b202320e350a4c0f26808f5e1fdR25